src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetWorkTypesControllerCFTTest.java

```diff
@@ -0,0 +1,124 @@
+import io.restassured.response.Response;
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import uk.gov.hmcts.reform.wataskmanagementapi.SpringBootFunctionalBaseTest;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.SystemDateProvider.DATE_TIME_FORMAT;
+
+public class GetWorkTypesControllerCFTTest extends SpringBootFunctionalBaseTest {
+
+    private static final String ENDPOINT_BEING_TESTED = "work-types";
+
+    private Headers authenticationHeaders;
+
+    @Before
+    public void setUp() {
+        authenticationHeaders = authorizationHeadersProvider
+            .getTribunalCaseworkerAAuthorization("wa-ft-test-r2-");
+    }
+
+    @Test
+    public void should_return_work_types_when_user_has_work_types() {
+        common.setupOrganisationalRoleAssignmentWithWorkTypes(authenticationHeaders);
+
+        Response result = restApiActions.get(
+            ENDPOINT_BEING_TESTED + "/?filter-by-user=true",
+            authenticationHeaders
+        );
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE);
+
+        final List<Map<String, String>> workTypes = result.jsonPath().getList(
+            "work_types");
+
+        List<Map<String, String>> expectedWarnings = Lists.list(
+            Map.of("id", "hearing_work", "label", "Hearing work"),
+            Map.of("id", "upper_tribunal", "label", "Upper Tribunal"),
+            Map.of("id", "routine_work", "label", "Routine work")
+        );
+        Assertions.assertEquals(expectedWarnings, workTypes);
+
+    }
+
+    @Test
+    public void should_return_empty_work_types_when_user_has_no_work_types() {
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        Response result = restApiActions.get(
+            ENDPOINT_BEING_TESTED + "/?filter-by-user=true",
+            authenticationHeaders
+        );
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE);
+
+        final List<Map<String, String>> workTypes = result.jsonPath().getList(
+            "work_types");
+
+        Assertions.assertTrue(workTypes.isEmpty());
+
+    }
+
+    @Test
+    public void should_return_all_work_types() {
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        Response result = restApiActions.get(
+            ENDPOINT_BEING_TESTED + "/?filter-by-user=false",
```

Comment:
just clarifying in terms of behaviour and requirement- if filter by user is false, we default to returning all the workTypes?

Reply:
yes, that is the business req